src/bootstrap-table.js

```diff
@@ -1538,7 +1538,7 @@ class BootstrapTable {
     // show no records
     if (!hasTr) {
       this.$body.html(`<tr class="no-records-found">${Utils.sprintf('<td colspan="%s">%s</td>',
-        this.getVisibleFields().length,
+        this.getVisibleColumnCount(),
```

Comment:
```
this.getVisibleFields().length +
  (Utils.hasDetailViewIcon(this.options) && this.options.detailViewAlign !== 'right' ? 1 : 0)

Reply:
I agree we can use the Utils.hasDetailViewIcon method, but we should use a function as wrapper for that.
If we have in the future new features which must be also checked, the code will be ugly.
So we can prevent it right now.

What do you think @wenzhixin

Reply:
OK, let me updated this PR.